File: Array/1578. Minimum Time to Make Rope Colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        i, j = 0, 0
        res = 0
        
        while i < len(neededTime) and j < len(neededTime):
            curr_total, curr_max = 0, 0
            
            while j < len(neededTime) and colors[i] == colors[j]:
                curr_total += neededTime[j]
                curr_max = max(curr_max, neededTime[j])
                j += 1
                
            res += (curr_total - curr_max)
            i = j
            
        return res

File: Array/1578. Minimum Time to Make Rope Colorful.c
#define MAX(a,b) ((a>b)?a:b)

int minCost(char * colors, int* neededTime, int neededTimeSize){
    int i = 0;
    int j = 0;
    int res = 0;
    
    while (i < neededTimeSize && j < neededTimeSize){
        int curr_sum = 0;
        int curr_max = 0;
        
        while (j < neededTimeSize && colors[i] == colors[j]){
            curr_sum += neededTime[j];
            curr_max = MAX(curr_max, neededTime[j]);
            j += 1;
        }
        
        res += (curr_sum - curr_max);
        i = j;
    }
    
    return res;
}
